contracts/lease/src/position/position_spec.rs

```diff
@@ -0,0 +1,121 @@
+use currency::Currency;
+use finance::{
+    coin::Coin,
+    liability::Liability,
+    price::{self, Price},
+};
+
+use crate::{
+    api::PositionSpecDTO,
+    error::{ContractError, ContractResult},
+};
+
+#[cfg_attr(test, derive(Debug))]
+pub struct PositionSpec<Lpn> {
+    liability: Liability,
+    min_asset: Coin<Lpn>,
+    min_trasaction_amount: Coin<Lpn>,
+}
+
+impl<Lpn> PositionSpec<Lpn>
+where
+    Lpn: Currency,
+{
+    fn new_internal(
+        liability: Liability,
+        min_asset: Coin<Lpn>,
+        min_trasaction_amount: Coin<Lpn>,
+    ) -> Self {
+        let obj = Self {
+            liability,
+            min_asset,
+            min_trasaction_amount,
+        };
+        debug_assert_eq!(Ok(()), obj.invariant_held());
+        obj
+    }
+
+    pub fn try_from(spec: PositionSpecDTO) -> ContractResult<Self> {
```

Comment:
we never want the domain to depend on DTOs! Please recall the hexagonal architecture